src/ngx_http_upsync_module.c

```diff
@@ -461,22 +463,57 @@ ngx_http_upsync_server(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
 
     ngx_memzero(&u, sizeof(ngx_url_t));
 
-    p = (u_char *)ngx_strchr(value[1].data, '/');
+    p = (u_char *)ngx_strchr(value[1].data, '@');
+    if (p == NULL) {
+        s.data = value[1].data;
+        s.len = value[1].len;
+
+        upscf->upsync_basic_auth.len = 0;
+        upscf->upsync_basic_auth.data = ngx_pcalloc(cf->pool, 1);
+    }
+    else {
+        size_t      len;
+        ngx_str_t   encoded, decoded;
+        u_char      *h;
+
+        s.data = p + 1;
+        s.len = value[1].len - (p + 1 - value[1].data);
+
+        /* Authorization: Basic base64(userpass) */
+        decoded.len = p - value[1].data;
+        decoded.data = value[1].data;
+
+        len = sizeof("Authorization: Basic ") - 1 + ngx_base64_encoded_length(decoded.len) + sizeof("\r\n");
+        h = ngx_pnalloc(cf->pool, len);
+        if (h == NULL) {
+            ngx_conf_log_error(NGX_LOG_EMERG, cf, 0, "upsync_server: out of memory");
+            return NGX_CONF_ERROR;
+        }
+
+        encoded.data = ngx_cpymem(h, "Authorization: Basic ", sizeof("Authorization: Basic ") - 1);
+        ngx_encode_base64(&encoded, &decoded);
+        (void)ngx_cpymem(encoded.data + encoded.len, "\r\n", sizeof("\r\n"));
```

Comment:
"\r\n" which is not the authorization part should not be here? If not use authorization, '\r\n' will not show, that is not right.